src/Serializers/BaseSerializer.ts

```diff
@@ -17,8 +17,9 @@ import { ISerializer } from "./ISerializer";
 export abstract class BaseSerializer<S, T extends IModelData, R = T> implements ISerializer<S, T, R> {
   public readonly ModelClass: IModelFactory<T>;
 
-  public abstract deserialize(data: R): IModel<T>;
-  public abstract serialize(modelData: IModel<T> | Partial<T>): S;
+  public abstract async deserialize(data: R): Promise<IModel<T>>;
+
```

Comment:
Any reason that we added this additional space?

Reply:
I think my IDE automatically adds an additional new line around abstract methods. I'll remove it.